src/wallet/wallet.cpp

```diff
@@ -2883,6 +2883,9 @@ std::shared_ptr<CWallet> CWallet::Create(WalletContext& context, const std::stri
     // TODO: Can't use std::make_shared because we need a custom deleter but
     // should be possible to use std::allocate_shared.
     std::shared_ptr<CWallet> walletInstance(new CWallet(chain, name, args, std::move(database)), ReleaseWallet);
+    walletInstance->m_keypool_size = std::max(args.GetIntArg("-keypool", DEFAULT_KEYPOOL_SIZE), (int64_t) 1);
```

Comment:
There might be a good reason for this, but why not initialize this member field in the header?

Reply:
where in the header exactly? constructor?

Reply:
Yes:
```
         : m_args(args),
           m_chain(chain),
           m_name(name),
-          m_database(std::move(database))
+          m_database(std::move(database)),
+          m_keypool_size(std::max(args.GetIntArg("-keypool", DEFAULT_KEYPOOL_SIZE), (int64_t) 1))
```

Reply:
Structurally speaking, the wallet class shouldn't have access to the global `ArgsManager` class, its internal behavior shouldn't be coupled to a global command line args parsing object. So setting this on the constructor would just hide the non-desire coupling.

Could actually go further here and totally remove the `m_args` member from the wallet now. Just need to keep track of the "-walletnotify" script.
Let me see how big this change is.

Reply:
Updated the PR with this changes.

Reply:
Great, that seems conceptually sound to me!